### PR TITLE
Added functionality to spin up local docker instance

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-699

### Change description ###

- Added functionality to spin up CCD stack with all dependent services locally.

- Added scripts to import CCD Definitions and utilities related to Idam to create roles, users  etc.

- Tested locally

- TODO

1. Update README.

2. Add queue emulator and publish docker image for bulk scanning orchestrator to hmcts registry.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: bin/create-user.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -e
+
+USER_EMAIL="${1:-me@server.net}"
+FORENAME="${2:-John}"
+SURNAME="${3:-Smith}"
+PASSWORD=Password12
+USER_GROUP="${4:-bsp-systemupdate}"
```
Comment (timja) [inline bin/create-user.sh]: is this a user group that exists?

File: bin/start-ccd-web.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+COMPOSE_FILES="-f docker-compose.yml"
+docker-compose ${COMPOSE_FILES} up ${@} -d ccd-case-management-web \
+                                           ccd-api-gateway \
+                                           idam-api \
+                                           authentication-web \
+                                           smtp-server
```
Comment (timja) [inline bin/start-ccd-web.sh]: is smtp server needed here?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        - IDAM_BSO_SYSTEMUPDATE_PASSWORD=Password12
+      mem_limit: 512m
+      memswap_limit: 0
+      depends_on:
+        shared-database-v10:
+          condition: service_healthy
+      ports:
+        - 8080:8080
+
+    smtp-server:
```
Comment (timja) [inline docker-compose.yml]: is this needed?

Comment (nitinprabhu) [reply docker-compose.yml]: `idam-api` needs smtp server `-SPRING_MAIL_HOST=smtp-server` else idam-api errors out

File: bin/create-user.sh (hunk 1 above)
Comment (nitinprabhu) [reply bin/create-user.sh]: yes `bsp-systemupdate` is configured in idam :)

File: bin/start-ccd-web.sh (hunk 2 above)
Comment (nitinprabhu) [reply bin/start-ccd-web.sh]: Reason I added it here is because to spin up docker container for smtp server which is then used by idam-api

**Comment by nitinprabhu** [comment]
Closing this one as the docker code will be moved to shared infra.

**Pull request closed without merging**